src/ui/public/state_management/state_storage/_tests__/state_hash.js

```diff
@@ -0,0 +1,49 @@
+import expect from 'expect.js';
+import sinon from 'sinon';
+
+import {
+  createStateHash,
+  isStateHash,
+} from '../state_hash';
+
+describe('stateHash', () => {
+  const existingJsonProvider = () => null;
+
+  describe('#createStateHash', () => {
+
+    describe('returns a hash', () => {
+      const json = JSON.stringify({a: 'a'});
+      const hash = createStateHash(json, existingJsonProvider);
+      expect(isStateHash(hash)).to.be(true);
+    });
+
+    describe('returns the same hash for the same input', () => {
+      const json = JSON.stringify({a: 'a'});
+      const hash1 = createStateHash(json, existingJsonProvider);
+      const hash2 = createStateHash(json, existingJsonProvider);
+      expect(hash1).to.equal(hash2);
+    });
+
+    describe('returns a different hash for different input', () => {
+      const json1 = JSON.stringify({a: 'a'});
+      const hash1 = createStateHash(json1, existingJsonProvider);
+
+      const json2 = JSON.stringify({a: 'b'});
+      const hash2 = createStateHash(json2, existingJsonProvider);
+      expect(hash1).to.not.equal(hash2);
+    });
+  });
+
+  describe('#isStateHash', () => {
+    it('returns true for values created using #createStateHash', () => {
+      const json = JSON.stringify({a: 'a'});
+      const hash = createStateHash(json, existingJsonProvider);
+      expect(isStateHash(hash)).to.be(true);
+    });
+
+    it('returns false for values not created using #createStateHash', () => {
```

Comment:
I recommend testing against rison too, as "hash vs. rison" it is the primary test that `isStateHash()` **must** be pass.